src/utils/FishingUtils.ts

```diff
@@ -0,0 +1,106 @@
+import { FSHTank } from '../import';
+import {
+  Profile,
+  Extension,
+  Logical,
+  Resource,
+  Instance,
+  FshValueSet,
+  FshCodeSystem,
+  Invariant,
+  RuleSet,
+  Mapping
+} from '../fshtypes';
+import { getVersionFromFshDefinition } from '../fhirtypes/common';
+import { logger } from './FSHLogger';
+import { Fishable, Metadata, Type } from './Fishable';
+
+// Use the provided fisher to fish for the item, which might have a |version appended.
+// If found, return it. If not found, try to remove any |version info and fish for any available version.
+export function fishForFHIRBestVersion(
+  fisher: Fishable,
+  item: string,
+  ...types: Type[]
+): any | undefined {
+  let result = fisher.fishForFHIR(item, ...types);
+
+  // If the item was not found, try slicing off a version and searching for any version
+  if (result == null && item != null && item.includes('|')) {
+    const [base, ...versionParts] = item.split('|');
+    const version = versionParts.join('|') || null;
+    result = fisher.fishForFHIR(base, ...types);
+    if (version != null && result?.version != null && version != result?.version) {
+      logger.warn(
+        `The ${base} definition was specified with version ${version}, but SUSHI found version ${result.version}`
```

Comment:
I feel like the language is not quite right here because I initially interpreted "The ${base} definition was specified with version ${version}" to mean it was _defined_ with version ${version}. But what we really mean is it was _requested_ with that version. So maybe:

> The ${base} definition was requested with version ${version}, but SUSHI found version ${result.version}

or even just putting the original value with the `|` in it, e.g.:

> ${item} was requested, but SUSHI found ${base}|${result.version}

Reply:
I like the simple second version!

Reply:
Simple second version updated in Part of [18b8ba8](https://github.com/FHIR/sushi/pull/1275/commits/18b8ba89e0c6aa3a9ec12bfc0c288154ec6be7f5). Let me know if you prefer something else!